### PR TITLE
Logging smithy interface and macros

### PR DESCRIPTION
**Pull request by stevelr**

### CODE REVIEW

File: logging/logging.smithy [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/// WriteLog - log a text message
+///
+operation WriteLog {
+    input: LogEntry
+}
+
+structure LogEntry {
+    /// severity level: debug,info,warn,error
+    @required
+    level: String,
```
Comment (brooksmtownsend) [inline logging/logging.smithy:38]: Can we also include trace here?

File: logging/rust/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+pub use wasm_log::log;
+
+#[cfg(target_arch = "wasm32")]
+mod wasm_log {
+    use std::string::ToString;
+    use wasmbus_rpc::RpcResult;
+
+    /// log a text message at a severity level (debug, info, warn, or error)
+    /// parameters may be `&str`, `String`, or `&String`.
+    /// You can also use this crate's macros `debug`,`info`,`warn`, and `error`
```
Comment (brooksmtownsend) [inline logging/rust/src/lib.rs:22]: Also just marking this comment if `trace` is added to edit here

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
Just a request to add trace level logging as an option

File: logging/logging.smithy (hunk 1 above)
Comment (autodidaddict) [reply logging/logging.smithy:38]: Elixir host doesn't have both debug and trace, only debug.

**Review by autodidaddict — APPROVED** [review]

File: logging/rust/src/lib.rs (hunk 2 above)
Comment (stevelr) [reply logging/rust/src/lib.rs:22]: I used the levels that were implemented in the elixir host.

**Pull request merged**